### PR TITLE
feat(zk_toolbox): Add SQL format for zk supervisor

### PR DESCRIPTION
**Pull request by matias-gonz**
## What ❔
Add SQL format for zk supervisor

### CODE REVIEW

File: zk_toolbox/crates/zk_supervisor/Cargo.toml [hunk 1]
```diff
@@ -29,3 +29,4 @@ futures.workspace = true
 types.workspace = true
 serde_yaml.workspace = true
 zksync_basic_types.workspace = true
+sqruff-lib = { git = "https://github.com/quarylabs/sqruff", version = "0.18.2" }
```
Comment (perekopskiy) [inline zk_toolbox/crates/zk_supervisor/Cargo.toml:32]: any reason to not use crate from crates.io?

Comment (perekopskiy) [reply zk_toolbox/crates/zk_supervisor/Cargo.toml:32]: also, did you consider https://crates.io/crates/sqlformat? It has much more downloads

Comment (matias-gonz) [reply zk_toolbox/crates/zk_supervisor/Cargo.toml:32]: Yes! They still didn't publish 1.18.2 (I already asked on their slack channel) so I had to import it like that.
I did try sqlformat but we weren't happy with the formatted queries.

**Review by perekopskiy — APPROVED** [review]

File: zk_toolbox/crates/zk_supervisor/src/commands/sql_fmt.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    let base_indent = line
+        .find(|c: char| !c.is_whitespace())
+        .context(MSG_FAILED_TO_DETERMINE_BASE_INDENT)?
+        + 4;
+
+    let prefix = &line[..query_start];
+
+    let query_end = if is_raw_string {
+        line.find(r#""#)
```
Comment (slowli) [inline zk_toolbox/crates/zk_supervisor/src/commands/sql_fmt.rs:90]: Does it work like that? IIUC, `r#""#` is an empty string; you'd probably want `"\"#"`.

File: core/lib/dal/src/blocks_dal.rs [hunk 3]
```diff
@@ -986,14 +986,21 @@ impl BlocksDal<'_, '_> {
         sqlx::query!(
             r#"
             INSERT INTO
```
Comment (slowli) [inline core/lib/dal/src/blocks_dal.rs:988]: Another example of lenient formatting: the formatter would accept here

```rust
        sqlx::query!(
            r#"
            INSERT INTO commitments (
                l1_batch_number,
                events_queue_commitment,
                bootloader_initial_content_commitment
            )
            VALUES ($1, $2, $3)
            ON CONFLICT (l1_batch_number) DO NOTHING
            "#,
            _
        )
```

or even a monstrosity like

```rust
        sqlx::query!(
            r#"
            INSERT
            INTO commitments (
                l1_batch_number, events_queue_commitment,
                bootloader_initial_content_commitment
            )
            VALUES (
                $1, $2, $3
            )
            ON CONFLICT
            (l1_batch_number)
            DO NOTHING
            "#,
            _
        )
```

**Review by slowli — COMMENTED** [review]
I don't quite like the new formatter because it looks overly lenient. For example, take this mangled query from `events_dal`:

```rust
        let logs: Vec<_> = sqlx::query_as!(
            StorageWeb3Log,
            r#"
            SELECT  address,
                topic1,
                topic2,
              topic3,
                topic4, value,
                NULL::bytea AS "block_hash",
                NULL::bigint AS "l1_batch_number?",
                miniblock_number ,
                tx_hash,
                   tx_index_in_block,
                event_index_in_block,
                event_index_in_tx, NULL::BIGint AS "block_timestamp?"
            FROM events
            WHERE
                tx_hash  = ANY     ( $1 )
            ORDER  BY
                miniblock_number ASC, event_index_in_block ASC
            "#,
            &hashes[..] as &[&[u8]],
        )
```

Its formatted variant is

```rust
        let logs: Vec<_> = sqlx::query_as!(
            StorageWeb3Log,
            r#"
            SELECT
                address,
                topic1,
                topic2,
                topic3,
                topic4, value,
                NULL::bytea AS "block_hash",
                NULL::bigint AS "l1_batch_number?",
                miniblock_number,
                tx_hash,
                tx_index_in_block,
                event_index_in_block,
                event_index_in_tx, NULL::bigint AS "block_timestamp?"
            FROM events
            WHERE
                tx_hash = ANY($1)
            ORDER BY
                miniblock_number ASC, event_index_in_block ASC
            "#,
            &hashes[..] as &[&[u8]],
        )
```

That is, the formatter doesn't have an opinion on how many list items are put on a single line. Obviously, having *any* sort-of reasonable formatter is better than not having one, but maybe it may be tweaked to be more opinionated?

File: core/lib/dal/src/blocks_dal.rs (hunk 3 above)
Comment (Deniallugo) [reply core/lib/dal/src/blocks_dal.rs:988]: But, i'd say the original option was even worse, no?

**Comment by Deniallugo** [comment]
@slowli Actually it support pretty big configuration https://docs.sqlfluff.com/en/1.3.0/configuration.html

we are free to set it more precise

**Comment by matias-gonz** [comment]
@slowli @Deniallugo 
Actually these are the docs:
https://github.com/quarylabs/sqruff/blob/main/docs/rules.md

**Review by Deniallugo — APPROVED** [review]

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**